src/base/vesting/vesting.ts

```diff
@@ -37,50 +46,88 @@ export async function withdrawVested(
   );
   const signer = wallet.signer;
 
-  state.set("withdrawingSign");
+  state.set({ type: "withdrawingSign" });
+
+  try {
+    const tx: TransactionResponse = await contract
+      .connect(signer)
+      .withdrawVested();
 
-  const tx = await contract.connect(signer).withdrawVested();
+    state.set({ type: "withdrawing", tx });
+    await tx.wait();
+  } catch (e: any) {
+    console.log(e);
 
-  state.set("withdrawing");
-  await tx.wait();
+    state.set({ type: "error", error: e.reason });
+    return;
+  }
   session.state.refreshBalance(wallet);
-  state.set("withdrawn");
+  state.set({ type: "withdrawn" });
 }
 
 export async function getInfo(
   address: string,
   wallet: Wallet,
-): Promise<VestingInfo> {
-  const contract = new ethers.Contract(
-    address,
-    ethereumContractAbis.vesting,
-    wallet.provider,
-  );
-  const token = await contract.token();
-  const beneficiary = await contract.beneficiary();
-  const withdrawable = await contract.withdrawableBalance();
-  const withdrawn = await contract.withdrawn();
-  const total = await contract.totalVestingAmount();
-  const vestingStartTime = await contract.vestingStartTime();
-  const vestingPeriod = await contract.vestingPeriod();
-  const cliffPeriod = await contract.cliffPeriod();
-
-  const tokenContract = new ethers.Contract(
-    token,
-    ethereumContractAbis.token,
-    wallet.provider,
-  );
-  const symbol = await tokenContract.symbol();
+): Promise<VestingInfo | undefined> {
+  const contract = getVestingContract(address, wallet);
+
+  let vestingInfo:
+    | [string, string, BigNumber, BigNumber, BigNumber, string, string, string]
+    | undefined = undefined;
+  let token: string | undefined = undefined;
+
+  try {
+    token = (await contract.token()) as string;
```

Comment:
Shouldn't we handle the `undefined` case here instead of asserting that it's always a `string`?

<img width="243" alt="Screenshot 2022-12-23 at 09 59 12" src="https://user-images.githubusercontent.com/158411/209305648-8f3cb68c-7e53-4a60-ba7b-520768317a3e.png">